src/ssl/gadgets.cc

```diff
@@ -479,18 +479,24 @@ static bool buildCertificate(Security::CertPointer & cert, Ssl::CertificatePrope
         (void)replaceCommonName(cert, properties.commonName);
     }
 
-    // We should get caCert notBefore and notAfter fields and do not allow
-    // notBefore/notAfter values from certToMimic before/after notBefore/notAfter
-    // fields from caCert.
-    // Currently there is not any way in openssl tollkit to compare two ASN1_TIME
-    // objects.
+    ASN1_TIME *notBefore = NULL;
+    ASN1_TIME *notAfter = NULL;
+
+    // Get times from cacert
+    notBefore = X509_getm_notBefore(properties.signWithX509.get());
+    notAfter = X509_getm_notAfter(properties.signWithX509.get());
```

Comment:
Here and elsewhere in this review, please see change requests for `notBefore`. They are symmetric.

This particular assignment can go inside the same `if` statement as the assignment for `caNotBefore`.